Formula/xwin.rb

```diff
@@ -0,0 +1,17 @@
+class Xwin < Formula
+  desc "Microsoft CRT and Windows SDK headers and libraries loader"
+  homepage "https://github.com/Jake-Shadle/xwin"
+  url "https://github.com/Jake-Shadle/xwin/archive/refs/tags/0.2.7.tar.gz"
+  sha256 "c5122f14113de20258bc156dc8912e37ccc8a214175c72e296bf14a25df61cf9"
+  license "Apache-2.0"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system bin/"xwin", "--accept-license", "splat", "--disable-symlinks"
```

Comment:
Can we check some output here?

Reply:
```ruby
  test do
    system bin/"xwin", "--accept-license", "splat", "--disable-symlinks", "--include-debug-libs", "--include-debug-symbols"
    assert_predicate testpath/".xwin-cache", :exist?
    (testpath/"test.cpp").write <<~EOS
      #include <cstdio>
      int main() {printf("Hello world!");return 0;}
    EOS

    system ENV.cc, "--target=x86_64-windows-msvc", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/shared", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/ucrt", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/um", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/winrt", \
      "-isystem", testpath/".xwin-cache/splat/crt/include", \
      "-Xclang", \
      "--dependent-lib=msvcrt", \
      "-MD", \
      "-MT", "test.obj", \
      "-MF", "test.obj.d", \
      "-o", "test.obj", \
      "-c", "test.cpp"
    assert_predicate testpath/"test.obj", :exist?

  end
```

I'm trying to compile simple program, but it fails from brew test:
```
brew test --keep-tmp xwin                                                                                               INT ✘  base   04:36:41 PM  
==> Testing xwin
==> /opt/homebrew/Cellar/xwin/0.2.7/bin/xwin --accept-license splat --disable-symlinks --include-debug-libs --include-debug-symbols
==> /usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/spl
Last 15 lines from /Users/zeka/Library/Logs/Homebrew/xwin/test.02.clang:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/_stdio.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/sys/_types/_va_list.h:32:26: error: typedef redefinition with different types ('__darwin_va_list' (aka 'void *') vs 'char *')
typedef __darwin_va_list va_list;
                         ^
/private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include/vadefs.h:72:23: note: previous definition is here
        typedef char* va_list;
                      ^
In file included from test.cpp:1:
/private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include/cstdio:38:13: error: no member named '_Mbstatet' in the global namespace; did you mean '__mbstate_t'?
using _CSTD _Mbstatet;
      ~~~~~ ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/i386/_types.h:81:3: note: '__mbstate_t' declared here
} __mbstate_t;
  ^
1 warning and 4 errors generated.
==> Temporary files retained at:
/private/tmp/xwin-test-20220830-95728-jjyqv0
Error: xwin: failed
An exception occurred within a child process:
  BuildError: Failed executing: /usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/ucrt -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/um -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/winrt -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include -Xclang --dependent-lib=msvcrt -MD -MT test.obj -MF test.obj.d -o test.obj -c test.cpp
/opt/homebrew/Library/Homebrew/formula.rb:2518:in `block in system'
/opt/homebrew/Library/Homebrew/formula.rb:2454:in `open'
/opt/homebrew/Library/Homebrew/formula.rb:2454:in `system'
/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/xwin.rb:22:in `block in <class:Xwin>'
/opt/homebrew/Library/Homebrew/formula.rb:2294:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/utils.rb:605:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:2293:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:959:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:2292:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `run'
/opt/homebrew/Library/Homebrew/formula.rb:2569:in `mktemp'
/opt/homebrew/Library/Homebrew/formula.rb:2286:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

while simple command works fine, without any error:
```sh
/usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/ucrt -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/um -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/winrt -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/crt/include -Xclang --dependent-lib=msvcrt -MD -MT test.obj -MF test.obj.d -o test.obj -c test.cpp
```

Reply:
Just added library existence check for now